Add progressViewOffset on iOS

scarlac opened the pull request:
Fixes #10718. This adds support for the progressViewOffset on the
RefreshControl component on iOS.

ericnakagawa commented:
@scarlac Hey Seph, is there anyway we could add some tests around these changes?

scarlac commented:
@ericnakagawa I get where you're going though I'm not sure how to go about it.

I could maybe make an assertion so that we're guaranteed that we're repositioning "something" but it wouldn't guarantee it would be the spinner. What's your opinion on that?

mmmulani commented on React/Views/RCTRefreshControl.m:
```diff
@@ -31,6 +32,18 @@ - (instancetype)init
+  // UIRefreshControl is managed so avoid changing it directly
+  // This method may break in future versions of iOS (subview structure may change)
+  // And self.transform won't shift title text
+  if(_isInitialRender) {
```
what view are we looking for? can we be more certain that it's the right one.
also I would a better explanation as to what view you're looking for so that whomever has to fix this when it breaks knows what to do :)

scarlac replied:
Great suggestion, @mmmulani! I actually forgot about that I could inspect the UI layers.
I'm thinking of this, which seems to work:
```
NSString *subviewClass = NSStringFromClass([subview class]);
if([subviewClass isEqualToString:@"_UIRefreshControlModernContentView"]) { ... }
```
I will look into adding a test instead of doing this runtime this week.

hramos replied:
Taking a look at this PR again. I'm concerned about the code here. It's been a while since I've worked on iOS and I am not sure about what current best practices are. @mmmulani do you approve of this method for finding the progress view subview?

hramos replied:
Also, cc @brentvatne since it sounded like you were interested in this PR. Can you take a look?

hramos commented:
Waiting on tests.

scarlac commented:
Where can I find some info on writing tests for React Native?

hramos commented:
I suggest taking a look at the existing set of tests. They can be found within the Examples/UIExplorer app.

mkonicek commented:
Should this PR also contain docs for the new prop?

ptomasroos commented:
@scarlac can you update this pr? Otherwise I can fork it and fix the last details?

scarlac commented:
@mkonicek: I've updated the docs  - the prop already exists, so I've just removed the "@platform android" tag.
@ptomasroos: I won't have the time to write an integration test for this. I'll have to leave it up to you to do that, if necessary.
Thanks for the follow up guys!

ptomasroos commented:
I dont see why would need integration tests for this, I guess a test plan with added examples with progressViewOffset in the UIExplorer would do, right @hramos @mkonicek ?

hramos commented:
That works for me.

arsen commented:
Looks like it is breaking it, it is not showing refresh indicator while you are scrolling down.
![refreshcontrol](https://cloud.githubusercontent.com/assets/3446647/24694730/e633d198-1997-11e7-9b10-dfa75ada8ca2.gif)

Weakky commented:
Are there any news about that PR guys ? I'm having the same issue as @arsen here.

scarlac commented:
@Weakky See my comment on #10718 - I don't have the time to fix it at the moment, sorry
